### PR TITLE
📖 Update link for investigations directory

### PR DESCRIPTION
**Pull request by cychiang**
## Summary
The link for `investigations directory` in [docs website](https://docs.kcp.io/kcp/main/en/GOALS/#process) is pointing to a 404 page. It might be good to point to the investigations folder in GitHub instead of 404 page.

### CODE REVIEW

**Comment by ncdc** [comment]
Thanks for catching this! Instead of linking to GitHub, how about deleting this sentence entirely? Thanks!

**Comment by cychiang** [comment]
This `The [investigations directory](https://github.com/kcp-dev/kcp/tree/main/docs/content/en/investigations) is where we will capture specific deep dive details.` entire sentence? Even better I think, those topics in the investigations directory seem been mentioned in the `Key Concepts` section except [self-service-policy](https://github.com/kcp-dev/kcp/blob/main/docs/content/en/investigations/self-service-policy.md). Should I ignore it for now?

File: docs/content/en/GOALS.md [hunk 1]
```diff
@@ -116,7 +116,7 @@ Not every idea below may bear fruit, but it's never the wrong time to look for n
 
 ### Process
 
-Right now we are interested in assessing how these goals fit within the larger ecosystem. The [investigations directory](docs/investigations) is where we will capture specific deep dive details.
+Right now we are interested in assessing how these goals fit within the larger ecosystem. The [investigations directory](https://github.com/kcp-dev/kcp/tree/main/docs/content/en/investigations) is where we will capture specific deep dive details.
```
Comment (ncdc) [inline docs/content/en/GOALS.md:119]:
```suggestion
Right now we are interested in assessing how these goals fit within the larger ecosystem.
```

**Review by ncdc — APPROVED** [review]

**Pull request merged**